src/python/aqueduct_executor/operators/connectors/data/execute.py

```diff
@@ -48,11 +48,49 @@ def run(spec: Spec) -> None:
     Arguments:
     - spec: The spec provided for this operator.
     """
+    return run_helper(
+        spec=spec,
+        read_artifacts_func=utils.read_artifacts,
+        write_artifact_func=utils.write_artifact,
+        setup_connector_func=setup_connector,
+        is_spark=False,
+    )
+
+
+def run_helper(
+    spec: Spec,
+    read_artifacts_func: Any,
+    write_artifact_func: Any,
+    setup_connector_func: Any,
+    is_spark: bool,
```

Comment:
If we have to pass this arg around, does it makes sense to explicitly pass a single `Optional[spark.Session]` object to decide if spark is enabled? Also I'd like to remove `**kwargs` and since it's not clear what to expect and how it's used. This pattern is more useful in cases like decorators where we support arbitrary inputs, which is not the case here

Reply:
This is the correct pattern, however it requires that we import `pyspark.sql` in the regular code path, which we want to avoid. The kwargs pattern was to avoid the import in non-Spark environments.